### PR TITLE
Add api that returns definitions by termid

### PR DESCRIPTION
**Pull request by gimochi**
## Related Issues
#110 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (one that would cause existing functionality to change)

## Testing
- [ ] Tested in some way, if yes, the description follows.

### CODE REVIEW

**Comment by enginehenryed** [comment]
~Add route that get definitions by termid~
Add Route that returns definitions ...
- Route returns definitions, not gets definitions

Add definition getting functionality

**Comment by enginehenryed** [comment]
Assign yourself

File: src/controllers/Definition/DefinitionGetController.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+export async function getDefinitionByTermId(req) {
+  const termSelected = await TermSelectDAO.selectTermByTermId(req.termId);
+
+  if (termSelected.length == 0) {
+    throw new MarmoymError(ErrorType.Term.TERM_NOT_FOUND);
+  } else {
+    termSelected[0]['defs'] = [];
+    //TODO 현재는 최신순으로 등록된 defs를 가져오지만 추후 vote순으로 가져오기
+    const defs = await DefinitionSelectDAO.selectRecentlyCreatedDefinitionsByTermId(req.termId, req.offset, 5);
```
Comment (enginehenryed) [inline src/controllers/Definition/DefinitionGetController.ts]: Why don't you try the such as following
```javascript
const defs = 
  await DefinitionSelectDAO.selectRecentlyCreatedDefinitionsByTermId(req.termId, req.offset, 5);
```

File: src/controllers/Definition/DefinitionGetController.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    await Promise.all(defs.map(async defObj => {
+      //get Pos 
+      const posSelected = await PosSelectDAO.selectPosByDefinitionId(defObj.id);
+      defObj['poss'] = [];
+      await Promise.all(posSelected.map(posObj => {
```
Comment (enginehenryed) [inline src/controllers/Definition/DefinitionGetController.ts]: Why posObj? Can it be just pos?

same as in def, usage, origin ,,.

File: src/dao/Term/TermSelectDAO.ts [hunk 3]
```diff
@@ -24,3 +24,11 @@ export function selectRecentlyUpdatedTerm(offset: number) {
     .limit(5)
     .offset(Number(offset));
 };
+
+export function selectTermByTermId(termId: number) {
+  return db('Term').select('Term.id  as termId', 'Term.label as term', 'Term.roman as termRoman')
+  .where({
```
Comment (enginehenryed) [inline src/dao/Term/TermSelectDAO.ts]:
```javascript
db().select()
  .where({
    ...
  })
 ```

File: src/controllers/Definition/DefinitionGetController.ts (hunk 2 above)
Comment (gimochi) [reply src/controllers/Definition/DefinitionGetController.ts]: My style

File: src/controllers/Definition/DefinitionGetController.ts (hunk 1 above)
Comment (gimochi) [reply src/controllers/Definition/DefinitionGetController.ts]: Ok

File: src/dao/Term/TermSelectDAO.ts (hunk 3 above)
Comment (gimochi) [reply src/dao/Term/TermSelectDAO.ts]: Ok

File: src/controllers/Definition/DefinitionGetController.ts (hunk 2 above)
Comment (enginehenryed) [reply src/controllers/Definition/DefinitionGetController.ts]: If you insist, I wouldn't complain. But know that putting a type in the variable name is discouraged. We habve TypeScript, in that case.

**Pull request merged**